### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by tectiv3**
Added some clarity to the folder structure
Cleaned cli commands
Moved mapping.json into the cloud
Simplified configurator
This is a preparation stage for #69

### CODE REVIEW

File: device/config/configurator.go [hunk 1]
```diff
@@ -1,93 +1,86 @@
...
+		Address string
+		Gateway string
+		Netmask string
+		DNS     string
+	}
+)
 
-	s := make(map[string]interface{})
+// New creates an empty Configurator
+func New(ssh ssh_helper.Util) *configurator {
```
Comment (dbsparkle) [inline device/config/configurator.go]: __[golint]__
* [golint] exported func New returns unexported type *config.configurator, which can be annoying to use

File: device/config/configurator.go [hunk 2]
```diff
@@ -1,93 +1,86 @@
...
 
-	s["ssh"] = ssh
+	storage["ssh"] = ssh
 
-	return &configurator{s, c}
+	return &configurator{storage, order}
+}
+
+// NewDefault creates a default Configurator
+func NewDefault(ssh ssh_helper.Util) *configurator {
```
Comment (dbsparkle) [inline device/config/configurator.go]: __[golint]__
* [golint] exported func NewDefault returns unexported type *config.configurator, which can be annoying to use

File: device/config/configurator.go [hunk 3]
```diff
@@ -1,93 +1,86 @@
...
+	config.order[Wifi] = NewCallbackFn(SetWifi, SaveWifi)
+	config.order[Interface] = NewCallbackFn(SetInterface, SaveInterface)
+	config.order[DNS] = NewCallbackFn(SetSecondaryDNS, SaveSecondaryDNS)
+	return config
 }
 
 // NewCallbackFn creates a new CallbackFn with 2 Function parameters
-func NewCallbackFn(config Function, apply Function) *CallbackFn {
-	return &CallbackFn{config, apply}
+func NewCallbackFn(config cb, apply cb) *callbackFn {
```
Comment (dbsparkle) [inline device/config/configurator.go]: __[golint]__
* [golint] exported func NewCallbackFn returns unexported type *config.callbackFn, which can be annoying to use

File: device/config/configurator.go [hunk 4]
```diff
@@ -110,28 +103,296 @@ func (c *configurator) Write() error {
...
+
+	_, eut, err := ssh.Run(fmt.Sprintf(`echo "%s" >> %s`, command, fp))
+	if err != nil {
+		return errors.New(err.Error() + ":" + eut)
+	}
+
+	return nil
+}
+
+// SetInterfaces is a set of dialog to set user `Interfaces`
```
Comment (dbsparkle) [inline device/config/configurator.go]: __[golint]__
* [golint] comment on exported function AskInterfaceParams should be of the form "AskInterfaceParams ..."

File: device/config/constants.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
+package config
+
+const (
+	// Default Configurator constants that are describe a specific configuration option
```
Comment (dbsparkle) [inline device/config/constants.go]: __[golint]__
* [golint] comment on exported const Locale should be of the form "Locale ..."

File: device/config/constants.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	Locale    = "Locale"
+	Keymap    = "Keymap"
```
Comment (dbsparkle) [inline device/config/constants.go]: __[golint]__
* [golint] exported const Keymap should have comment (or a comment on this block) or be unexported

File: device/config/constants.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+	MountDir = "/tmp/isaax-sd/"
+
+	Language   = "LANGUAGE=%s\n"
+	LocaleAll  = "LC_ALL=%s\n"
+	LocaleLang = "LANG=%s\n"
+
+	DefaultLocale = "en_US.UTF-8"
+
+	ISAAX_CONF_DIR = "/etc/"
```
Comment (dbsparkle) [inline device/config/constants.go]: __[golint]__
* [golint] don't use ALL_CAPS in Go names; use CamelCase

File: device/config/constants.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+	TMP_DIR        = "/tmp/"
```
Comment (dbsparkle) [inline device/config/constants.go]: __[golint]__
* [golint] don't use ALL_CAPS in Go names; use CamelCase

File: lib/repo/mapping.go [hunk 9]
```diff
@@ -82,6 +85,17 @@ func (d *deviceCollection) findDevice(device string) (*DeviceMapping, error) {
 	return nil, errors.New(missingRepo)
 }
 
+func DownloadDevicesRepository() {
```
Comment (dbsparkle) [inline lib/repo/mapping.go]: __[golint]__
* [golint] exported function DownloadDevicesRepository should have comment or be unexported

File: workstation/w_darwin.go [hunk 10]
```diff
@@ -8,40 +8,26 @@ import (
...
+	return &workstation{runtime.GOOS, true, m, ms}
 }
 
 const diskSelectionTries = 3
 const writeAttempts = 3
 
-// Notifies user to chose a mount, after that it tries to write the data with `diskSelectionTries` number of retries
-func (d *darwin) WriteToDisk(img string) (job *help.BackgroundJob, err error) {
+// Notifies user to choose a mount, after that it tries to write the data with `diskSelectionTries` number of retries
+func (d *workstation) WriteToDisk(img string) (job *help.BackgroundJob, err error) {
```
Comment (dbsparkle) [inline workstation/w_darwin.go:30]: __[gometalinter]__
* [gocyclo] cyclomatic complexity 21 of function (*workstation).WriteToDisk() is high (> 15)

File: iotit.go [hunk 11]
```diff
@@ -1,204 +1,181 @@
...
+			},
+		},
+		{
+			Name:    "install",
+			Aliases: []string{"i"},
+			Usage:   "Install to global app environment",
+			Action: func(c *cli.Context) error {
+				log.Debug("Checking ", installPath, progName)
+				if _, err := os.Stat(installPath + progName); os.IsNotExist(err) {
+					p, err := os.Executable()
```
Comment (dbsparkle) [inline iotit.go:77]: __[gometalinter]__
* [interfacer] Executable not declared by package os
* [staticcheck] Executable not declared by package os
* [unconvert] Executable not declared by package os
* [gosimple] Executable not declared by package os

Comment (tectiv3) [reply iotit.go:77]: I think we need to update go to 1.8 in sideci. Can we do that?

File: workstation/w_darwin.go (hunk 10 above)
Comment (tectiv3) [reply workstation/w_darwin.go:30]: so sad

File: device/config/configurator.go [hunk 12]
```diff
@@ -110,28 +102,296 @@ func (c *configurator) Write() error {
...
 
 // RemoveConfigFn removes CallbackFn from order
-func (c *configurator) RemoveConfigFn(num int) {
-	if num == len(c.order)-1 {
-		c.order = c.order[:num]
+func (c *Configurator) RemoveConfigFn(name string) {
+	delete(c.order, name)
+}
+
+func (c *Configurator) StoreValue(name string, value interface{}) {
```
Comment (dbsparkle) [inline device/config/configurator.go:124]: __[golint]__
* [golint] exported method Configurator.StoreValue should have comment or be unexported

File: iotit.go (hunk 11 above)
Comment (dbsparkle) [inline iotit.go:77]: __[gometalinter]__
* [unconvert] Executable not declared by package os
* [interfacer] Executable not declared by package os
* [staticcheck] Executable not declared by package os
* [gosimple] Executable not declared by package os

**Review by kylycht — APPROVED** [review]

**Pull request merged**